ios/FluentUI/SegmentedControl/SegmentedControl.swift

```diff
@@ -545,4 +585,39 @@ open class SegmentedControl: UIView, TokenizedControlInternal {
                                                                         index + 1, items.count)
         }
     }
+
+    private func updateStackDistribution() {
+        stackView.distribution = shouldSetEqualWidthForSegments ? .fillEqually : .fillProportionally
+    }
+
+    private func updateGradientMaskColors() {
+        gradientMaskLayer.colors = gradientMaskColors
+    }
+
+    private func updateViewHierarchyForScrolling() {
+        if isFixedWidth {
+            scrollView.removeFromSuperview()
+            pillContainerView.removeFromSuperview()
+            addSubview(pillContainerView)
+            layer.mask = nil
+
+            NSLayoutConstraint.deactivate(scrollViewConstraints)
+        } else {
+            pillContainerView.removeFromSuperview()
+            scrollView.addSubview(pillContainerView)
+            addSubview(scrollView)
+            layer.mask = gradientMaskLayer
+
+            NSLayoutConstraint.activate(scrollViewConstraints)
+        }
+    }
+
+    private var maximumContentOffset: CGFloat {
+        return stackView.frame.size.width - scrollView.frame.size.width
+    }
+
+    // MARK: UIScrollViewDelegate
+    public func scrollViewDidScroll(_ scrollView: UIScrollView) {
```

Comment:
nit: it should be closer to other public functions

Reply:
Moved to an extension, which seems to be a common pattern in our code. Will follow up with the team in general to see which approach should be preferred in the future.